nbgitpuller/pull.py

```diff
@@ -303,11 +303,13 @@ def main():
 
     parser = argparse.ArgumentParser(description='Synchronizes a github repository with a local repository.')
     parser.add_argument('git_url', help='Url of the repo to sync')
+    parser.add_argument('branch_name', default='master', help='Branch of repo to sync', nargs='?')
```

Comment:
Should the default branch to pull remain `master`, or should the default branch to pull be the default branch of the repo? 

I'm not sure if it is technically possible, but from my intuition the default behavior should similar to doing a `git clone <repo>` where you simply get the default branch of the repo checked out. Would it be technically possible to let the argument branch_name have a default value of a blank string and not hardcode it to the master branch?

Reply:
This is a somewhat sloppy review as I have not read up on the gitpuller CLI much - but the key point I'm after is: can we avoid hardcoding "master" as default if the argument isn't specified?

Reply:
Yeah, we should try and autodetect the default branch same way we do in the web interface

Reply:
Yes this is no problem--should have not hard coded master. If there is no branch passed i can determine the default branch of the repo.